### PR TITLE
Workaround config

### PR DESCRIPTION
**Pull request by BraisGabin**
Right now we are using `FailFastConfig` and `DisabledAutoCorrectConfig` to modify the configuration of all detekt and not just the configuration related with the rules itself.

This is kind of complex because we have all those configurations mixed. We should think about this for 2.0.

I need this refactor to implement `--all-rules`. Because I want to enable all the rules, but I don't want to enable any PostProcessor or Report. And all of them checks the same key: `"active"`.

I really don't like too much this code but I don't want to change any current behaviour.

### CODE REVIEW

**Review by arturbosch — COMMENTED** [review]
I like this!

I was thinking if this location is the only one which needs to get wrapped for fail fast and co (instead of creating a new settings objects which potentially can lead to opening two threadpool/classloaders/environments)?

https://github.com/detekt/detekt/blob/7328075ae6adbb064637c5eb417e5a6f5d750c06/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/Analyzer.kt#L27

**Comment by BraisGabin** [comment]
I think that we don't need to wrap it any where else.

File: detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/tooling/Lifecycle.kt [hunk 1]
```diff
@@ -57,13 +60,36 @@ internal interface Lifecycle {
...
         { generateBindingContext(settings.environment, settings.classpath, it) },
     override val processorsProvider: () -> List<FileProcessListener> =
         { FileProcessorLocator(settings).load() },
     override val ruleSetsProvider: () -> List<RuleSetProvider> =
-        { spec.rulesSpec.runPolicy.createRuleProviders(settings) }
+        { settings.createRuleProviders() }
 ) : Lifecycle
+
+private fun workaroundConfiguration(settings: ProcessingSettings): ProcessingSettings {
+    return ProcessingSettings(settings.spec, settings.spec.workaroundConfiguration(settings.config))
```
Comment (arturbosch) [inline detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/tooling/Lifecycle.kt:74]: I mean that `settings.spec.workaroundConfiguration(settings.config)` could just be in `Analyzer` and we do not need to create a new `ProcessingSettings` at all. Or do I miss something?

Comment (arturbosch) [reply detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/tooling/Lifecycle.kt:74]: Replacing this line

https://github.com/detekt/detekt/blob/7328075ae6adbb064637c5eb417e5a6f5d750c06/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/Analyzer.kt#L27

to `private val config: Config = settings.spec.workaroundConfiguration(settings.config)`

Comment (BraisGabin) [reply detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/tooling/Lifecycle.kt]: Oh! Yes, you are completely right. It have more sense to do it in the `Analyzer`. Done.

**Review by arturbosch — APPROVED** [review]

**Pull request merged**